src/axelrod_dojo/algorithms/genetic_algorithm.py

```diff
@@ -77,22 +84,57 @@ def crossover(population, num_variants):
 
     def evolve(self):
         self.generation += 1
-        print("Scoring Generation {}".format(self.generation))
+        if self.print_output:
+            print("Scoring Generation {}".format(self.generation))
 
         # Score population
         scores = self.score_all()
         results = list(zip(scores, range(len(scores))))
         results.sort(key=itemgetter(0), reverse=True)
 
-        # Report
-        print("Generation", self.generation, "| Best Score:", results[0][0],
-              repr(self.population[results[0][1]]))
         # Write the data
-        row = [self.generation, mean(scores), pstdev(scores), results[0][0],
-               repr(self.population[results[0][1]])]
-        self.outputer.write(row)
-
-        ## Next Population
+        # Note: if using this for analysis, for reproducability it may be useful to alter the data passed back
+
+        opponent_strategy_name = 'unknown'
+
+        for strategy, init_kwargs in self.opponents_information:
+            opponent = strategy(**init_kwargs)
+            opponent_strategy_name = type(opponent).name
+            try:
+                player_seed = init_kwargs['seed']
+            except KeyError:
+                player_seed = None
+
+        # Row Data Cols:
+        # 0:Generation number
+        # 1:Population score mean
+        # 2:Population score median
+        # 3:Population Variance
+        # 4:Population Score range
+        # 5:Best result score (Highest score)
+        # 6:parameters of Best result
+        # 7:Opponent strategy name
+        # 8:Seed (if applicable, else None)
+
+        # Opponent Sequence can be reconstructed by playing a match against the exact seed params
+        row = [self.generation,
```

Comment:
So this is the reason the doctest is failing, you're outputting way more information then is currently in the doctest which is expecting a different output. If we want to keep this extra output then the doctest should maybe be run with `print_output=False` because the floats will continue to cause errors even if you wrote them in.

This would require changing the rest of that tutorial too though obviously (perhaps just explaining that the output has been written to a `training_output.csv`).